hafnian/_hermite_multidimensional.py

```diff
@@ -0,0 +1,135 @@
+    r"""
+    Returns the multidimensional Hermite polynomials :math:`H_k^{(R)}(y)`. Here :math:`R` is an :math:n \times n: square matrix,
+    :math:`y` is an :math:`n` dimensional vector. The polynomials are parametrized by the multi-index :math:`k=(k_0,k_1,\ldots,k_{n-1})
+    and are calculated for all values :math:`0 \leq k_j < \text{resolution}`, thus a tensor of dimensions :math:`\text{resolution}^n` is returned.
+    This tensor can either be flattened into a vector or returned as an actual tensor with :math:`n` indices.
+    Note that is R = np.array([[1.0+0.0j]]) then :math:`H_k^{(R)}(y)` are precisely the well known **probabilists' Hermite polynomials** :math:`He_k(y)`:,
+    and if R = 2*np.array([[1.0+0.0j]]) then :math:`H_k^{(R)}(y)` are precisely the well known **physicists' Hermite polynomials** :math:`H_k(y)`:.
+    Args:
+        R (array): Square matrix parametrizing the Hermite polynomial family
+        resolution (int): Maximum size of the subindices in the Hermite polynomial
+        y (array): Vector for the argument of the Hermite polynomial
+        renorm (bool): If True returns :math:`H_k^{(R)}(y)/\prod(\prod_i k_i!)`
+        make_tensor: If False returns a flattened one dimensional array instead of a tensor with the values of the polynomial
+    Returns:
+        (array): The multidimensional Hermite polynomials.
+    """
+    input_validation(R, check_symmetry=False)
+    n, _ = R.shape
+    if y is None:
+        y = np.zeros([n], dtype=complex)
+
+    m = y.shape[0]
+    if m != n:
+        raise ValueError("The matrix R and vector y have incompatible dimensions")
+
+    values = np.array(hm(R, y, resolution, ren=renorm))
+
+    if make_tensor:
+        shape = resolution * np.ones([n], dtype=int)
+        values = np.reshape(values, shape)
+
+    return values
+
+
+def hafnian_batched(A, resolution, mu=None, tol=1e-12, renorm=False, make_tensor=True):
+    r"""Returns the haf(reduction(A, k)) where k is a vector of (non-negative) integers with the same dimensions as the square matrix A
+    :math:`k = (k_0,k_1,\ldots,k_{n-1})` and where :math:`0 \leq k_j < \text{resolution}`.
+    If mu is not None the it instead calculates :math:`lhaf(fill_diagonal(reduction(A, k),reduction(mu, k)))`, this calculation can only be performed if
+    the matrix A has an inverse.
+    Args:
+        R (array): Square matrix parametrizing
+        resolution (int): Maximum size of the subindices in the Hermite polynomial
+        y (array): Vector for the argument of the Hermite polynomial
+        renorm (bool): If True returns :math:`haf(reduction(A, k))/\prod(\prod_i k_i!)` or :math:`lhaf(fill_diagonal(reduction(A, k),reduction(mu, k)))` is mu is not None
+        make_tensor: If False returns a flattened one dimensional array instead of a tensor with the values of the polynomial
+    Returns:
+        (array): The values of the hafnians.
+    """
```

Comment:
same comment as above re: line breaks. You can build the docs locally (`make doc`) to ensure the docstrings render correctly :)